runtime/src/main/java/com/flipkart/flux/controller/WorkFlowExecutionController.java

```diff
@@ -240,6 +243,9 @@ public void redriveTask(Long taskId) {
         if(isTaskRedrivable(state.getStatus()) && state.getAttemptedNoOfRetries() < state.getRetryCount()) {
             logger.info("Redriving a task with Id: {} for state machine: {}", state.getId(), state.getStateMachineId());
             executeStates(state.getStateMachineId(), Collections.singleton(state));
+        } else {
+            //cleanup the tasks which can't be redrived from redriver db
+            this.redriverRegistry.deRegisterTask(taskId);
         }
     }
 
```

Comment:
We should deregister even in the case when updateExecutionStates is called with  fatal "errored" state since we don't want that to be redriven.